src/EventStore.Core/LogV3/LogV3RecordFactory.cs

```diff
@@ -5,6 +5,12 @@
 namespace EventStore.Core.LogV3 {
 	public class LogV3RecordFactory : IRecordFactory<long> {
 		public LogV3RecordFactory() {
+			if (!BitConverter.IsLittleEndian) {
```

Comment:
isn't BinaryWriter (what we use to write/read log records to/from disk) little endian by default? (which should be independent of the memory endianness)

Reply:
ah nevermind, got it - it's due to the structs

Reply:
yeah the v3 records are using the binary writer, but not field by field. we're just giving it the whole span of memory for the record. there is similar 'islittleendian' check in the grpc code so it looks like only little endian is supported at the moment anyway, so we decided that if we want to have big-endian later we can address it then. the way we'd support big endian is by adding some translation so that the files are always littleendian, so we aren't producing invalid files in the mean time